Easydict/Feature/ViewController/Cell/EZTableTipsCell.m

```diff
@@ -0,0 +1,269 @@
+//
+
+#import "EZTableTipsCell.h"
+#import "EZOpenLinkButton.h"
+#import "NSImage+EZSymbolmage.h"
+#import "EZLanguageManager.h"
+
+@interface EZTableTipsCell()
+@property (nonatomic, strong) NSView *contentView;
+@property (nonatomic, strong) NSImageView *tipsIconImageView;
+@property (nonatomic, strong) NSTextField *tipsNameLabel;
+@property (nonatomic, strong) NSTextField *tipsContentLabel;
+@property (nonatomic, strong) EZOpenLinkButton *moreBtn;
+@property (nonatomic, strong) EZOpenLinkButton *solveBtn;
+@property (nonatomic, strong) NSDictionary *dataDict;
+@property (nonatomic, strong) NSString *questionSolveURL;
+@property (nonatomic, strong) NSString *seeMoreURL;
+@end
+
+@implementation EZTableTipsCell
+
+- (instancetype)initWithFrame:(NSRect)frame
+{
+    self = [super initWithFrame:frame];
+    if (self) {
+        [self setupUI];
+        [self updateQuestionContent];
+    }
+    return self;
+}
+
+- (void)setupUI {
+    // mas key
+    self.contentView.mas_key = @"topBarView";
+    self.tipsIconImageView.mas_key = @"tipsIconImageView";
+    self.tipsNameLabel.mas_key = @"tipsNameLabel";
+    self.moreBtn.mas_key = @"moreBtn";
+    self.solveBtn.mas_key = @"solveBtn";
+    
+    CGSize iconSize = CGSizeMake(20, 20);
+    
+    // constraints
+    [self.contentView mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.edges.mas_equalTo(0);
+    }];
+    
+    [self.tipsIconImageView mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.left.mas_equalTo(9);
+        make.top.mas_equalTo(9);
+        make.size.mas_equalTo(iconSize);
+    }];
+    
+    [self.tipsNameLabel mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.left.mas_equalTo(self.tipsIconImageView.mas_right).offset(6);
+        make.centerY.mas_equalTo(self.tipsIconImageView.mas_centerY).offset(1);
+    }];
+    
+    [self.solveBtn mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.left.mas_equalTo(9);
+        make.top.mas_equalTo(self.tipsContentLabel.mas_bottom).offset(9);
+        make.size.mas_equalTo(CGSizeMake(98, 32));
+    }];
+    
+    [self.moreBtn mas_makeConstraints:^(MASConstraintMaker *make) {
+        make.left.mas_equalTo(self.solveBtn.mas_right).offset(20);
+        make.top.mas_equalTo(self.solveBtn);
+        make.size.mas_equalTo(CGSizeMake(94, 32));
+    }];
+}
+
+- (void)updateConstraints {
+    
+    [self.tipsContentLabel mas_updateConstraints:^(MASConstraintMaker *make) {
+        make.left.mas_equalTo(9);
+        make.width.mas_lessThanOrEqualTo(self.bounds.size.width - 9);
+        make.top.mas_equalTo(self.tipsNameLabel.mas_bottom).offset(12);
+    }];
+    
+    [super updateConstraints];
+}
+
+- (void)updateQuestionContent {
+    NSArray *questions = self.dataDict[@"questions"];
+    int index = arc4random() % questions.count;
+    self.tipsContentLabel.stringValue = questions[index];
+    NSArray *solves;
+    if ([[EZLanguageManager shared].ezCurrentLanguage isEqualToString:@"zh-Hans"]) {
```

Comment:
Do not use the system language to compare directly, the value sometimes ends with an area code, such as `zh-Hans-CN`.

<img width="1095" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/abea4ec7-f166-4ba3-aa47-1f7564e23167">

Reply:
You can use `systemFirstLanguage` instead.

```objc
@property (nonatomic, copy) EZLanguage systemFirstLanguage;
```

Reply:
Just use `isSystemChineseFirstLanguage` to check if user system language is Chinese.